bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ForecastWeatherHandler.java

```diff
@@ -0,0 +1,220 @@
+import java.time.Instant;
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.fmiweather.internal.client.Data;
+import org.openhab.binding.fmiweather.internal.client.FMIResponse;
+import org.openhab.binding.fmiweather.internal.client.ForecastRequest;
+import org.openhab.binding.fmiweather.internal.client.LatLon;
+import org.openhab.binding.fmiweather.internal.client.Location;
+import org.openhab.binding.fmiweather.internal.client.exception.FMIResponseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ForecastWeatherHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Sami Salonen - Initial contribution
+ */
+@NonNullByDefault
+public class ForecastWeatherHandler extends AbstractWeatherHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ForecastWeatherHandler.class);
+
+    private static final String GROUP_FORECAST_NOW = "forecastNow";
+    private static final int QUERY_RESOLUTION_MINUTES = 20; // The channel group hours should be divisible by this
+    // Hirlam horizon is 54h https://ilmatieteenlaitos.fi/avoin-data-avattavat-aineistot (in Finnish)
+    private static final int FORECAST_HORIZON_HOURS = 50; // should be divisible by QUERY_RESOLUTION_MINUTES
+    private static final Map<String, Map.Entry<String, @Nullable Unit<?>>> CHANNEL_TO_FORECAST_FIELD_NAME_AND_UNIT = new HashMap<>(
+            9);
+
+    private static void addMapping(String channelId, String requestField, @Nullable Unit<?> unit) {
+        CHANNEL_TO_FORECAST_FIELD_NAME_AND_UNIT.put(channelId,
+                new AbstractMap.SimpleImmutableEntry<>(requestField, unit));
+    }
+
+    static {
+        addMapping(CHANNEL_TEMPERATURE, PARAM_TEMPERATURE, CELSIUS);
+        addMapping(CHANNEL_HUMIDITY, PARAM_HUMIDITY, PERCENT);
+        addMapping(CHANNEL_WIND_DIRECTION, PARAM_WIND_DIRECTION, DEGREE_ANGLE);
+        addMapping(CHANNEL_WIND_SPEED, PARAM_WIND_SPEED, METRE_PER_SECOND);
+        addMapping(CHANNEL_GUST, PARAM_WIND_GUST, METRE_PER_SECOND);
+        addMapping(CHANNEL_PRESSURE, PARAM_PRESSURE, MILLIBAR);
+        addMapping(CHANNEL_PRECIPITATION_INTENSITY, PARAM_PRECIPITATION_1H, MILLIMETRE_PER_HOUR);
+        addMapping(CHANNEL_TOTAL_CLOUD_COVER, PARAM_TOTAL_CLOUD_COVER, PERCENT);
+        addMapping(CHANNEL_FORECAST_WEATHER_ID, PARAM_WEATHER_SYMBOL, null);
+    }
+
+    private @NonNullByDefault({}) LatLon location;
```

Comment:
I have understood that the way it is know is recommended by the [conventions](https://www.openhab.org/v2.4/docs/developer/development/conventions.html): 

> Fields within ThingHandler classes that are initialized within the initialize() method may also be annotated like this, because the framework ensures that initialize() will be called before any other method.